optuna/samplers/_tpe/sampler.py

```diff
@@ -457,15 +472,33 @@ def sample_independent(
 
         return self._sample(study, trial, {param_name: param_distribution})[param_name]
 
+    def _get_params(self, trial: FrozenTrial) -> dict[str, Any]:
+        if trial.state.is_finished() or not self._multivariate:
```

Comment:
Please correct me if I'm wrong, but probably either of the following is better?

```suggestion
        if trial.state.is_finished() or not self._constant_liar:
```
or 

```suggestion
        if trial.state.is_finished() or not self._constant_liar or not self._multivariate:
```